webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeGroupRepository.scala

```diff
@@ -215,6 +221,12 @@ class RoLDAPNodeGroupRepository(
                        .entry2NodeGroup(x)
                        .toIO
                        .chainError(s"Error when mapping server group entry to its entity. Entry: ${sgEntry}")
+                       .flatMap(g => {
+                         ZIO
+                           .foreach(g.serverList)(nodeFactRepo.get _)
+                           .map(_.flatMap(_.map(_.id)))
+                           .map(nodeIds => g.copy(serverList = nodeIds))
+                       })
```

Comment:
I was wondering if it wasn't quicker to get nodeIds from nodeFact (keySet, already filtered for the tenant) and intersect with g.serverlist. My intuition is that it's better, but my perf intuition are generaly not good. WDYT?

Reply:
That is a clever idea ! I agree that it may be not so performant as it is : the `ZIO.foreach` on a set may be as slow as `O(n)`...